File: core/src/main/java/cz/vhromada/common/validator/MovableValidator.java
package cz.vhromada.common.validator;

import cz.vhromada.common.Movable;
import cz.vhromada.result.Result;

/**
 * An interface represents validator for movable data.
 *
 * @param <T> type of data
 * @author Vladimir Hromada
 */
public interface MovableValidator<T extends Movable> {

    /**
     * Validates data.
     *
     * @param data            validating data
     * @param validationTypes types of validation
     * @return result with validation errors
     */
    Result<Void> validate(T data, ValidationType... validationTypes);

}

File: common-core/src/main/kotlin/com/github/vhromada/common/validator/MovableValidator.kt
package com.github.vhromada.common.validator

import com.github.vhromada.common.entity.Movable
import com.github.vhromada.common.result.Result

/**
 * An interface represents validator for movable data.
 *
 * @param <T> type of data
 * @author Vladimir Hromada
 */
interface MovableValidator<T : Movable> {

    /**
     * Validates data.
     *
     * @param data            validating data
     * @param validationTypes types of validation
     * @return result with validation errors
     */
    fun validate(data: T?, vararg validationTypes: ValidationType): Result<Unit>

}
